pkg/apis/validation/validation.go

```diff
@@ -125,6 +120,37 @@ func keyEncodingValue(path *field.Path, s string) field.ErrorList {
 	return nil
 }
 
+// filename ensures that a given filename, is indeed a valid filename. It does
+// this by validating that the given filename is not:
+// 1. absolute
+// 2. starting with '..'
+// 3. contain '/'
+// 4. longer than 255 characters
+func filename(path *field.Path, filename string) field.ErrorList {
+	var el field.ErrorList
+
+	if filepath.IsAbs(filename) {
+		el = append(el, field.Invalid(path, filename, "filename must not be an absolute path"))
+	}
+
+	if filepath.HasPrefix(filename, "..") {
```

Comment:
**issue (blocking):** `filepath.HasPrefix` is deprecated: https://pkg.go.dev/path/filepath#HasPrefix

**suggestion:** `strings.HasPrefix` should do the trick

Reply:
Thanks @jahrlin, I've updated to use `strings.HasPrefix` as well as adding a validation check to ensure filenames must not start or end with spaces.